ntoskrnl/ke/amd64/freeze.c

```diff
@@ -46,6 +65,9 @@ KiProcessorFreezeHandler(
     /* We are running again now */
     CurrentPrcb->IpiFrozen = IPI_FROZEN_STATE_RUNNING;
 
+    /* This is for synchronization on VMs */
```

Comment:
Can you explain to us (commit message can be a good place) what's this "synchronization on VMs" is all about?

Reply:
A "short" explanation here (I'll include a better one into the commit message, and improve the comment): VMs do some "optimizations" to improve performance, including when it comes to cross-processor synchronization. Certain multiprocessor guarantees sometimes don't work correctly in VMs. Usually these are VM bugs, I wouldn't rule out hardware bug or me getting something wrong either though. The CPUID instruction is a synchronizing instruction on bare metal, it is a VM exit on a VM. This enforces memory visibilty on all CPUs, which on bare metal should be guaranteed by cache coherency protocols. I am currently not sure why exactly this doesn't work (on a VM), but possibly, because running under a hypervisor will relax some of these things and put the burdon of management on the hypervisor. I'll have another look to rule out that I missed something here.

Reply:
New testing doesn't seem to have the previous issue anymore. I removed it for now. If I see any issues coming up, I'll investigate and amend as needed.